mm2src/lp_ordermatch.rs

```diff
@@ -4351,6 +4411,20 @@ pub async fn create_maker_order(ctx: &MmArc, req: SetPriceReq) -> Result<MakerOr
         let mut my_maker_orders = ordermatch_ctx.my_maker_orders.lock();
         my_maker_orders.insert(new_order.uuid, Arc::new(AsyncMutex::new(new_order.clone())));
     }
+
+    let balance = match base_coin.my_spendable_balance().compat().await {
+        Ok(balance) => Some(balance),
+        Err(_) => None,
+    };
+
+    let balance_loop_exists = ordermatch_ctx.balance_loop_exists(base_coin.ticker());
+    if !balance_loop_exists {
```

Comment:
It seems that in some cases the loop might be not started when required. Let's imagine the following situation:
1. A node has many orders (e.g. 10k), one of them is for the RICK coin.
2. This order gets cancelled.
3. `coin_has_active_maker_orders` in the actually operates with a *copy* of `maker_orders` state. It's running for some time and will return `false`.
4.  While `coin_has_active_maker_orders` is running, user creates one order for RICK. During creation, it will not spawn `check_balance_update_loop` because `balance_loop_exists` is still true.
5. After this, `coin_has_active_maker_orders` completes in the loop, returns *false*, performs `remove_balance_loop` and stops the loop.

So we have one active order, but no background balance check for it.

To mitigate this, I propose to create the following (pseudocode):
```rust
struct MakerOrdersContext {
    orders: HashMap<Uuid, Arc<AsyncMutex<MakerOrder>>>,
    count_by_tickers: HashMap<String, usize>
}

impl MakerOrdersContext {
    fn add_order(&mut self, order: MakerOrder) {
         *self.count_by_tickers.entry(order.base).or_insert_with(0) += 1;
         self.orders.insert(order.uuid, Arc::new(AsyncMutex::new(order)));
    }
    ...
}
```

This will be guarded by a single `PaMutex` so the state will be always in sync. It will also allow avoiding iteration over all orders with async mutex lock on each.